### PR TITLE
Fix `bounds_error=True` ignored with 1D interpolation

### LINKED ISSUE #4854: `bounds_error=True` ignored in 1D interpolation
**What happened**:

Attempted to interpolate outside coordinate range while passing `bounds_error=True` through `kwargs` dictionary (which internally gets passed to `scipy.interpolate.interp1d`). Instead if `interp` function raising a `ValueError` nans are returned.

**What you expected to happen**:

I expected a `ValueError` exception to be raised when I've passed in the kwargs `bounds_error=True`.

**Minimal Complete Verifiable Example**:

```python
import xarray as xr
import numpy as np

da = xr.DataArray(
    np.sin(0.3 * np.arange(12).reshape(4, 3)),
    [("time", np.arange(4)), ("space", [0.1, 0.2, 0.3])],
)

# this should return nans, as the default is to fill with nans
da.interp(time=3.5)

>   <xarray.DataArray (space: 3)>
>   array([nan, nan, nan])
>   Coordinates:
>     * space    (space) float64 0.1 0.2 0.3
>       time     float64 3.5


# this should raise ValueError, but `bounds_error` is ignored
da.interp(time=3.5, kwargs=dict(bounds_error=True))

>   <xarray.DataArray (space: 3)>
>   array([nan, nan, nan])
>   Coordinates:
>     * space    (space) float64 0.1 0.2 0.3
>       time     float64 3.5

```

**Anything else we need to know?**:

I've made a pull-request to fix the issue

**Environment**:

<details><summary>Output of <tt>xr.show_versions()</tt></summary>

INSTALLED VERSIONS
------------------
commit: None
python: 2.7.15.final.0
python-bits: 64
OS: Linux
OS-release: 3.10.0-957.27.2.el7.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_GB.UTF-8
LOCALE: None.None

xarray: 0.10.7
pandas: 0.23.0
numpy: 1.14.3
scipy: 1.1.0
netCDF4: None
h5netcdf: None
h5py: 2.7.1
Nio: None
zarr: None
bottleneck: 1.2.1
cyordereddict: None
dask: 0.17.5
distributed: 1.21.8
matplotlib: None
cartopy: None
seaborn: None
setuptools: 39.1.0
pip: 10.0.1
conda: 4.6.4
pytest: 3.5.1
IPython: 5.7.0
sphinx: 1.7.4

</details>

### PR DESCRIPTION
**Pull request by leifdenby**
Previously `bounds_error` was always passed as `False` to
`scipy.interpolate.interp1d` and the value in `kwargs` was ignored
making it impossible to get bounds errors when doing 1D interpolation.

- [x] Closes #4854
- [x] Tests added
- [ ] Passes `pre-commit run --all-files`
- [ ] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [ ] New functions/methods are listed in `api.rst`

### CODE REVIEW

**Comment by dcherian** [comment]
Thanks @leifdenby  can you add a test for this behaviour please?

File: xarray/tests/test_interp.py [hunk 1]
```diff
@@ -866,3 +866,20 @@ def test_interpolate_chunk_advanced(method):
...
+@requires_scipy
+@pytest.mark.parametrize("bounds_error", [True, False])
+def test_interp1d_bounds_error(bounds_error):
+    """Ensure exception on bounds error is raised if requested"""
+    da = xr.DataArray(
+        np.sin(0.3 * np.arange(4)),
+        [("time", np.arange(4))],
+    )
+
+    if bounds_error:
```
Comment (dcherian) [inline xarray/tests/test_interp.py:880]: let's avoid the `parametrize` and `if` with

```
with pytest.raises(ValueError):
    da.interp(time=3.5, kwargs=dict(bounds_error=True)

# default is to fill with nans, so this should pass
da.interp(time=3.5, kwargs=dict(bounds_error=False))
```

**Comment by dcherian** [comment]
@leifdenby thanks, please add a line to whats-new.rst taking credit for this change if you'd like.

File: doc/whats-new.rst [hunk 2]
```diff
@@ -96,6 +96,7 @@ Bug fixes
   :py:meth:`Dataset.to_zarr` (:issue:`4783`, :pull:`4795`).
   By `Julien Seguinot <https://github.com/juseg>`_.
 - Add :py:meth:`Dataset.drop_isel` and :py:meth:`DataArray.drop_isel` (:issue:`4658`, :pull:`4819`). By `Daniel Mesejo <https://github.com/mesejo>`_.
+- Ensure that `Dataset.interp` raises `ValueError` when interpolating outside coordinate range and `bounds_error=True` (:issue:`4854`, :pull:`4855`). By `Leif Denby <https://github.com/leifdenby>`_.
```
Comment (dcherian) [inline doc/whats-new.rst:99]:
```suggestion
- Ensure that :py:meth:`Dataset.interp` raises ``ValueError`` when interpolating outside coordinate range and ``bounds_error=True`` (:issue:`4854`, :pull:`4855`). By `Leif Denby <https://github.com/leifdenby>`_.
```

**Comment by dcherian** [comment]
Thanks @leifdenby this looks great to me. I see this is your first PR. welcome to xarray!

**Review by dcherian — APPROVED** [review]

**Review by andersy005 — APPROVED** [review]

**Pull request merged**